controllers/repo_manager/ingress.go

```diff
@@ -158,28 +162,23 @@ func (r *RepoManagerReconciler) pulpIngressController(ctx context.Context, pulp
 }
 
 // pulp-ingress
-func (r *RepoManagerReconciler) pulpIngressObject(ctx context.Context, m *repomanagerv1alpha1.Pulp, plugins []IngressPlugin) *netv1.Ingress {
-	IsNginxIngressSupported := false
-	ingressClassList := &netv1.IngressClassList{}
-	ingressClassName := ""
-	if err := r.List(ctx, ingressClassList); err == nil {
-		for _, ic := range ingressClassList.Items {
-			ingressClassName = ic.Name
-			if ic.Spec.Controller == "k8s.io/ingress-nginx" {
-				IsNginxIngressSupported = true
-				break
-			}
```

Comment:
I think this should be the approach when the ingressClassName isn't provided

Reply:
And what should be the approach if there is no default ingressclass?
Or what if there is more than one default ingressclass?

Reply:
> And what should be the approach if there is no default ingressclass? 

IsNginxIngressSupported will return false
>Or what if there is more than one default ingressclass?

it will pick the name of the first Nginx IngressClass

Reply:
> "IsNginxIngressSupported will return false"

Hum... not sure if we are talking about the same thing, but I could not find in our current implementation the relationship between IsNginxIngressSupported and a default ingressclass.
In which part of the code we are checking the name of the default ingressclass?


> "it will pick the name of the first Nginx IngressClass"

Are you sure about that?
From what I could understand, it will pick the last class from a list (which is kind of a selection based on the list order) if a nginx controller is not found.
Why is it better to pick it "randomly" instead of letting users define which one they want to use?

Reply:
it gets the name here:
https://github.com/pulp/pulp-operator/blob/main/controllers/repo_manager/ingress.go#L167

> Why is it better to pick it "randomly" instead of letting users define which one they want to use?

I'm not against letting users defining it, I'm just saying the operator can pick one if a name isn't provided